src/lv_widgets/lv_cpicker.c

```diff
@@ -524,6 +527,13 @@ static void draw_disc_grad(lv_obj_t * cpicker, const lv_area_t * mask)
     area_mid.y2 -= inner;
 
     lv_draw_rect(&area_mid, mask, &bg_dsc);
+
+    /*Tidy outside of colour picker, the way the outer ring has to be drawn it cannot mask well so this draws a clean edge around it */
+    lv_draw_line_dsc_t tidy_dsc;
+    lv_draw_line_dsc_init(&tidy_dsc);
+    tidy_dsc.width = 3;
+    tidy_dsc.color = lv_obj_get_style_bg_color(cpicker, LV_CPICKER_PART_MAIN);
+    lv_draw_arc(cx, cy, r,  0, 360,  mask, &tidy_dsc);
```

Comment:
If I understand correctly, this means that if someone placed the color picker on top of another widget, there would now be a 3px wide halo around the color picker?

Reply:
Hi @embeddedt , 

Yes, it's not ideal!( but looks better than the ragged edge:smiley:)  I'll continue to think about how to deal with it and see if I can come up with something better if it's not satisfactory....

Kind Regards,

Pete

Reply:
I'd assume there must be a way to mask off a 3px wide ring around the outside instead of filling the pixels with a color, since that's how corners of objects are masked off. Maybe @kisvegabor has an idea.

Reply:
Hi @embeddedt & @kisvegabor ,

I think I have managed to fix this the correct way now!

Here is a sample output with the colour picker on a container with a black background:

![cpicker](https://user-images.githubusercontent.com/6610652/88377417-0c662d00-cd97-11ea-9015-1cf5ae2ae996.png)

I took some inspiration from the line meter widget.

Let me know if this is okay now and I'll get it merged to master.

Many Thanks.

Pete

Reply:
The implementation looks fine in general; the only thing I'm concerned about is the magic 3px number. Is there a way we can calculate that, or will it always be 3px off even if the color picker is twice the size?

Reply:
Hi @embeddedt,

I have tried it at various sizes and screen resolutions and all seems well.  The reason it's required is when the multicoloured lines are calculated for the outer ring in the loop their lengths are jittering because of the integer basis of the calculations and I printed out some values in the loop to check the delta and found that 2 was the worst case so a the 'magic 3' should cover all cases.  I could give it a local define name above the function with a description or add a comment about it something along the lines of the above if you feel it should be better explained.:smiley:

Cheers,

Pete

Reply:
Adding a local define at the top of the file sounds good to me.

Reply:
Okay I'll sort it and push the change...